ast3/Python/ast.c

```diff
@@ -1546,11 +1546,11 @@ ast_for_arguments(struct compiling *c, const node *n)
                     if (!vararg)
                         return NULL;
 
-                i += 2; /* the star and the name */
-                if (TYPE(CHILD(n, i)) == COMMA)
-                    i += 1; /* the comma, if present */
+                    i += 2; /* the star and the name */
+                    if (i < NCH(n) && TYPE(CHILD(n, i)) == COMMA)
+                        i += 1; /* the comma, if present */
 
-                    if (TYPE(CHILD(n, i)) == TYPE_COMMENT) {
+                    if (i < NCH(n) && TYPE(CHILD(n, i)) == TYPE_COMMENT) {
```

Comment:
@guido: It seems very likely to me that this error is the cause of the `Function has duplicate type signatures` you saw in #36.  A spurious type comment here (which could happen from the uninitialized read) would produce the correct error, and this only happens with a `*arg`, which lines up with what you saw.